Update plant.ino

wmurphy41 opened the pull request:
This is the code that I used in my version of the automatic plant watering system.  Main changes:

1. Used SoftwareSerial.h library to access the bluetooth module through pins 3 (TX) and 2 (RX).  This change fixed a problem where I could not load programs while the module was active.  Seemed to be a conflict on the serial bus.

2. Added a timeout feature so that the pump would turn off after 10 continual seconds so that it would not burn out when the reservoir runs dry.

3. Added additional commands for controlling the pump.  In addition to starting and stopping the pump, you can now adjust the moisture limit up and down, check the moisture limit, pulse the pump for 2 seconds, and reset the pump after the timeout.

4. Also moved code to C++ from C.

wmurphy41 committed:
Update plant.ino

jonathanrjpereira commented:
Hey William, 
Thanks for submitting these amazing changes and new features. I will merge this PR as soon as I make the necessary changes to the schematic & PCB files across all mediums.

wmurphy41 committed:
Update plant.ino

wmurphy41 committed:
Update plant.ino

- Updates to customization values
  o Lowered upper moisture limit to 1000

- Cleaned up use of constants.  They are now used in the initialization of global objects, not in the object definitions themselves

- Changes to control loop:
  o Added logic for when pump is in timeout mode
  o Added logic so that when when the moisture sensor is triggered, it waits an hour before being triggered again

Pump:
- Added function to tell if pump is in timeout mode
- Added function to tell if pump is currently running
-

wmurphy41 committed:
Update plant.ino

- Added safeMillis function to Pump to allow for Arduino clock rollover
- Added some more status messages to the Check function to show clock rollovers and to show when moisture sensor is offline.

wmurphy41 committed:
V2.  Capacitative Sensor

This version replaced the resistance moisture sensor with a capacitative sensor and added terminal connectors to the board for the moisture sensor, power, and pump.

wmurphy41 committed:
Cleanup timeout code

Changed the timeout code to be based  on stopping the pump once the reservoir is depleted.

RhysSlater commented:
Thanks for getting back to me, I since checked on your profile and it seems you have the same project with your updated coding? 

If this is the case I can use the code you produced on top of Jonathan’s original work.

wmurphy41 commented:
That's the case.  You're welcome to it.  Hope it's useful.

RhysSlater commented:
Hi, I am going to be using this code for my own project. However I am using the Arduino UNO. Are they only changes required for the analogue input and TX/RX pins? Thanks

wmurphy41 commented:
I believe so.  I think you can route the Tx/Rx to any two digital pins and the temp line to any analog pin.  Just update the constant.  Let me know if you run into any issues.

 

That reminds me I need to go rescue my planter from my office.  The Quarantine has outlasted the reservoir.

 

Bill

________________

William Murphy

 

From: Rhys Slater <notifications@github.com> 
Sent: Wednesday, April 1, 2020 1:00 PM
To: jonathanrjpereira/Automatic-Plant-Watering-System-using-Arduino <Automatic-Plant-Watering-System-using-Arduino@noreply.github.com>
Cc: William A. Murphy <wmurphy41@aol.com>; Author <author@noreply.github.com>
Subject: Re: [jonathanrjpereira/Automatic-Plant-Watering-System-using-Arduino] Update plant.ino (#1)

 

Hi, I am going to be using this code for my own project. However I am using the Arduino UNO. Are they only changes required for the analogue input and TX/RX pins? Thanks

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/jonathanrjpereira/Automatic-Plant-Watering-System-using-Arduino/pull/1#issuecomment-607370708> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFMGWOEVORRQHRYWDL5MDLDRKNXIHANCNFSM4H6OHMGA> .  <https://github.com/notifications/beacon/AFMGWOHD3IORY44BUL5K3STRKNXIHA5CNFSM4H6OHMGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEQZ33VA.gif>

wmurphy41 committed:
Cleanup messaging

- Minor cleanups to messages
- Increase baseline resevoir timeout to 90 seconds

wmurphy41 committed:
New mechanical design

New physical design acts as enclosure for electronics and water reservoir for use with any potted plant.

wmurphy41 committed:
Replace .heic files with .jpg

No other changes.
